pandas/indexes/multi.py

```diff
@@ -116,12 +116,22 @@ def __new__(cls, levels=None, labels=None, sortorder=None, names=None,
 
         return result
 
-    def _verify_integrity(self):
+    def _verify_integrity(self, new_labels=None, new_levels=None):
         """Raises ValueError if length of levels and labels don't match or any
-        label would exceed level bounds"""
+        label would exceed level bounds
+
+        Parameters
+        ----------
+        new_labels : optional list
+            Labels to check for validity. Defaults to current labels.
+        new_levels : optional list
+            Levels to check for validity. Defaults to current levels.
+        """
         # NOTE: Currently does not check, among other things, that cached
         # nlevels matches nor that sortorder matches actually sortorder.
-        labels, levels = self.labels, self.levels
+        labels = new_labels or self.labels
```

Comment:
Can `new_levels` or `new_labels` ever be a NumPy array (e.g. user passes `idx.set_levels(np.array['a', 'b'])`, or will they have been coerced to a list by now? If they're an array this will raise a ValueError. You could use `labels = new_labels if new_labels is not None else self.labels`.